cdp_scrapers/instances/kingcounty.py

```diff
@@ -0,0 +1,122 @@
+        """
+        Return URLs for videos and captions parsed from kingcounty.gov web page
+
+        Parameters
+        ----------
+        legistar_ev : Dict
+            Data for one Legistar Event obtained from
+            ..legistar_utils.get_legistar_events_for_timespan()
+
+        Returns
+        -------
+        List[Dict]
+            List of video and caption URI
+            [{"video_uri": ..., "caption_uri": ...}, ...]
+        """
+        try:
+            # a td tag with a certain id pattern containing url to video
+            
+            with urlopen(legistar_ev[LEGISTAR_EV_SITE_URL]) as resp:
+                soup = BeautifulSoup(resp.read(), "html.parser")
+            
+        except URLError or HTTPError:
+            log.debug(f"Failed to open {legistar_ev[LEGISTAR_EV_SITE_URL]}")
+            return []
+
+        try:
+            # this gets us the url for the web PAGE containing the video
+            
+            extract_video_url = soup.find(
+                "a",
+                id=re.compile(r"ct\S*_ContentPlaceHolder\S*_hypVideo"),
+                class_="videolink",
+            )["onclick"]
+            start= extract_video_url.find("'")+len("'")
+            end= extract_video_url.find("',")
+            video_page_url='https://kingcounty.legistar.com/'+extract_video_url[start:end]
+
+            
+        # catch if find() didn't find video web page url (no <a id=... href=.../>)
+        except KeyError:
+            log.debug("No URL for video page on {legistar_ev[LEGISTAR_EV_SITE_URL]}")
+            
+            return []
+
+        log.debug(f"{legistar_ev[LEGISTAR_EV_SITE_URL]} -> {video_page_url}")
+
+        try:
+            with urlopen(video_page_url) as resp:
+                # now load the page to get the actual video url
+                soup = BeautifulSoup(resp.read(), "html.parser")
+               
+                
+               
+        except URLError or HTTPError:
+            log.error(f"Failed to open {video_page_url}")
+            return []
+
+        # source link for the video is embedded in the script of downloadLinks. 
+        video_script_text = soup.find(
+            "script", text=re.compile(r"downloadLinks")
+        ).string
+        downloadLinks = video_script_text.split("[[")[1]
+        video_url_extract=downloadLinks.split(",\"")[1]
+        video_url=downloadLinks.split("\",")[1].strip('"')
+        video_url = video_url.replace("\\","")
+        
+        video_uris=[]
+        video_uris.append(video_url)
+        # caption URIs are not found for kingcounty events. keeping it to maintain the return format.
+        caption_uri=[]
+        iter = range(len(video_uris))
+        list_uri = []
+
+        for i in iter:
+            try:
+                video_uri = video_uris[i]
+            except IndexError:
+                video_uri = None
+
+            list_uri.append({CDP_VIDEO_URI: video_uri, CDP_CAPTION_URI: caption_uri})
```

Comment:
A bit confused as to what is happening here... why are you creating a _list_ of video uris... shouldn't there be only one?

Reply:
This is my "fault" really. The web page (seattlechannel.org) has a list of videos and captions for the player setup call, even though it [almost] always has just one of each. So, I think 99.9999% `get_video_uris()` will return something like
```
[{"video_uri" : "...", "caption_uri" : "..."}]
```
I wrote [the SeattleScraper one] it so that it _can_ return if it finds multiple.
```
[{"video_uri" : "...", "caption_uri" : "..."}, {"video_uri" : "...", "caption_uri" : "..."}, ...]
```

Reply:
Should we just have `get_video_uris()` return 1 `Dict` (or `NamedTuple` after change from above comment) of video and caption URI, instead of a `List[Dict]`? I think I still kind of like the current behavior, because it plays better with `EventIngestionModel.sessions` being a `List[Session]`, not `Session`. we create a `Session` per item from `get_video_uris()`.